frontend/src/components/pages/teacher/DisplayAssessmentResultsPage/DisplayAssessmentResultsByStudentPage.tsx

```diff
@@ -173,39 +32,74 @@ const STUDENT_STATISTICS_CONFIG: StatisticsConfig = {
 };
 
 const DisplayAssessmentResultsByStudentPage = () => {
-  const [selectedStudentId, setSelectedStudentId] = React.useState<string>();
+  const { sessionId } = useParams<{ sessionId: string }>();
+  const { data, error, loading } = useQuery<{
+    testSession: TestSessionWithResultsData;
+  }>(GET_TEST_SESSION_WITH_RESULTS, { variables: { id: sessionId } });
 
-  const students = MOCK_STUDENTS;
-  const idToStudentMap = React.useMemo(
+  const [selectedStudentId, setSelectedStudentId] = useState<string>("");
+
+  const studentsWithViewedState = useMemo(
+    () =>
+      data?.testSession.results?.map(({ student }) => ({
+        ...student,
+        // TODO @jfdoming: Replace this with the actual viewed state.
```

Comment:
is this work tracked somewhere? 👀

Reply:
Oops meant to make a ticket for this and another thing, my bad